### PR TITLE
[PATCH v7] build: support platforms that do not use config

### PR DESCRIPTION
**Pull request by lumag**
Support platforms that do not use config file.

Change-Id: I9f82337bcd547e0757cc1b5b0c9cdbcf99758b63
Signed-off-by: Dmitry Eremin-Solenikov <deremin-solenikov@cavium.com>

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -231,6 +231,8 @@ AS_IF([test "${with_platform}" = "linux-generic"],
 AC_DEFINE_UNQUOTED([ODP_IMPLEMENTATION_NAME], ["$ODP_IMPLEMENTATION_NAME"],
 		   [Define to the name of the implementation])
 
+AM_CONDITIONAL([ODP_USE_CONFIG], [test "xodp_use_config" = "true"])
```
Comment (matiaselo) [inline configure.ac]: Shouldn't the comparison be `[test "x$odp_use_config" = "xtrue"]`?

Comment (lumag) [reply configure.ac]: Hmm, true. I tested only for my platform, which doesn't use config. Sorry.

File: configure.ac [hunk 2]
```diff
@@ -231,6 +231,8 @@ AS_IF([test "${with_platform}" = "linux-generic"],
 AC_DEFINE_UNQUOTED([ODP_IMPLEMENTATION_NAME], ["$ODP_IMPLEMENTATION_NAME"],
 		   [Define to the name of the implementation])
 
+AM_CONDITIONAL([ODP_USE_CONFIG], [test "xodp_use_config" = "xtrue"])
```
Comment (matiaselo) [inline configure.ac]: The $ sign is still missing in front of `odp_use_config`.

Comment (lumag) [reply configure.ac]: fixed

**Comment by matiaselo** [comment]
Reviewed-by tags missing, otherwise ready to merge.

**Comment by lumag** [comment]
Unfortunately this is not enough: `EXTRA_DIST` in main `Makefile.am` references `config/odp-$(with_platform).conf`. If bundling several platforms, this variable will not be multi-substituted (for each platform), so the generated tarball will have one config file at most.
Updated PR with proposed solution.

**Comment by matiaselo** [comment]
@lumag Looks good, please add the reviewed-by tags from me and Stand, so I can merge this PR.

**Pull request merged**